### DISCUSSION TITLE
MotionEye which  Raspi camera to buy (on RockPi 4 ) ?

### DISCUSSION THREAD

**Discussion by jalsco** [MichaIng/DietPi#6416] [Q&A]
Am on latest version of DietPi v8.18
Already using MotionEye with one RTSP camera on local LAN, and **MotionEye is AWESOME !** 
**Which Raspi camera should I buy ? module 2.1 or module 3 ??**

There's some confusing discussion out there about pi cameras generally with DietPi..  I think the options have changed over the years.  
e.g. https://github.com/MichaIng/DietPi/issues/98   even that is quite old... 

There are reports that  pi camera v2.1 DOES works for sure on RockPi4. 
And then in Jan 2023 the  **NEW pi camera module 3** became available: https://www.raspberrypi.com/news/new-autofocus-camera-modules/  . Of course it did not even exist when much of that earlier internet discussion was written. 

The pi camera 3 seems much better - ie autofocus - but will it work with DietPi unning on RockPi ? 
And a more broader question, besides MotionEye, is the pi camera 3 even supported on DietPi at all ? 

I assume the pi camera (either module 2 or 3 ) will connect using this option in MotionEye: 

![image](https://github.com/MichaIng/DietPi/assets/7985459/2a0a1336-7764-415d-9a57-f50ad8fced78)

I know another option is to use a USB connected camera and one of these : 

![image](https://github.com/MichaIng/DietPi/assets/7985459/6b0f0606-2b98-42f9-b3ee-44acc0d717f8)

BUT - I have many reasons to prefer the actual pi camera, rather than a USB camera, in my current configuration... 

**Any advice, suggestions,  tips or comments about MotionEye, and specifically with the pi cameras, most greatly appreciated 🙏🏻**

**Comment by MichaIng** [comment]
To be true, I am not 100% sure whether the RPi Camera Module 3 will work with motionEye on ROCK Pi 4. On Raspberry Pi it is not trivial: https://github.com/motioneye-project/motioneye/issues/2683

It seems to not work as MMAL camera there (which requires the legacy graphics+camera stack to be enabled), but also not as common V4L2 camera (using e.g. `/dev/video0`), but only with a netcam wrapper, i.e. using some software which provides the camera stream locally as netcam stream, then adding this netcam in motionEye. motionEye does not support the libcamera tools yet, only MMAL (legacy RPi camera driver) and generic V4L2.

MMAL cannot work on ROCK Pi 4, as it is RPi-only. While libcamera tools are available on all devices, it is not supported by motionEye. And I am not sure whether it is an RPi-only issue with ribbon cable cameras and `/dev/videoX` devices, or whether this does not work on other SBCs with such port either. On RPi, if an RPi camera module is attached, there appear dozens of `/dev/videoX` devices, and none of them works, while the libcamera tools can successfully access the camera.

**Reply by jalsco** [reply]
Hi @MichaIng and thanks for your reply ! 😊

Thanks for including that link !  I had read it a few weeks ago, but at time I made the post I could not find that link to include in my post.  As always, and ALWAYS greatly appreciated, you have added extra information that it is critical to my question, thank you 🙏🏻🙏🏻

It seems from your answer that it would be safer for me to just BUY the **pi camera module 2.1** .   Are there any similar concerns about that camera working with MotionEye under Dietpi ??  Again - I assume (but I am now not confident in this assumption) - that the 2.1 module camera would  be setup as an MMAL camera on MotionEye ?

**Reply by jalsco** [reply]
ON same LAN I also have an actual Raspberry Pi 3b+ ( only running PiHole and running RasPi-OS ) , maybe I could attach the camera to that one and access it as a Network Camera from the RockPi ?   Seems a bit tricky, as it would not be a RTSP camera. If I need to install MotionEye on that old raspi , I won't go down that path. 

 I prefer to attach whatever camera I get directly to the RockPi4... I know the module 2 camera CAN attach and work (in theory) with the rockpi 4 : https://wiki.radxa.com/Rockpi4/MIPI_Camera   ..Does that mean it would be connected as a "MIPI Camera", whatever that is ... And are cameras like that "MIPI / CSI " ,  supported by MotionEye ? 

Raxda also has a camera https://docs.radxa.com/en/accessories/camera_4k . BUT I get the feeling ( not confirmed explicitly either way ), that it only works on Rock5 and later..

**Comment by jalsco** [comment]
Am planning to buy  the raspi camera 2.1 - it will, for certain, connect and be visible to Rockpi 4. 

Will it also work with MotionEye (under Dietpi) ? And what "camera type" should be configured as , in MotionEye ?   Based on comments above - I believe will be V4L2 type 🤷🏻‍♂️

**Reply by MichaIng** [reply]
Yes, as V4L2 camera on any other SBC than RPi. MMAL is an RPi only thing and only with legacy camera stack.

**Reply by jalsco** [reply]
I will let you know how the camera goes (in next couple weeks) 😊
Thanks for your reply 🙏🏻

**Comment by jalsco** [comment]
Hi @MichaIng  

Frustrating trying to make PiCamera work  with MotionEye/DietPi on my RockPi 4. Please help 🙏🏻

1. MotionEye is working and great for a couple months !  I have been using it with an RSTP camera.
2. Trying to add a PiCamera as "camera 2" I chose V4L2 and selected default camera, but when didn't work a tried again with each of those option in the dropdown... 
![image](https://github.com/MichaIng/DietPi/assets/7985459/75e10db3-c1c1-4923-9916-cb7d0957feeb)
3. In MotionEye the camera shows up , but always stays blank and shows no video 
![image](https://github.com/MichaIng/DietPi/assets/7985459/37643287-81b3-4f53-b665-c95afc97f680)

![image](https://github.com/MichaIng/DietPi/assets/7985459/ccf8bffe-d9ea-4a16-b03e-658145d7847a)


4. I tried to google for similar symptoms, and tried add bottom 2 lines to `/boot/armbianEnv.txt` , but it did nothing : 

```
verbosity=4
bootlogo=false
overlay_prefix=rockchip
overlays=i2c7 pcie-gen2
fdtfile=rockchip/rk3399-rock-pi-4b.dtb
rootdev=UUID=dfa3ab95-cede-4159-9841-4e31bafe853e
rootfstype=ext4
usbstoragequirks=0x2537:0x1066:u,0x2537:0x1068:u
docker_optimizations=on
console=none
camera_auto_detect=1
start_x=1
```


Any thoughts or advice on how to make the Camera work would be GREATLY appreciated ! 🙏🏻

**Reply by jalsco** [reply]
Also trying to make sense of this link and how I should add it into DietpPi 🤷🏻‍♂️

[https://wiki.radxa.com/Rockpi4/MIPI_Camera](https://wiki.radxa.com/Rockpi4/MIPI_Camera)

**Reply by jalsco** [reply]
@MichaIng  please.... any guidance at all ? 🙏🏻

**Comment by jalsco** [comment]
@MichaIng 

```
[Mon Oct 02] root@batopi: ~$ v4l2-ctl --list-devices
rockchip,rk3399-vpu-enc (platform: hantro-vpu):
        /dev/video0
        /dev/video1
        /dev/media1

rockchip-rga (platform:rga):
        /dev/video3

rkvdec (platform:rkvdec):
        /dev/video2
        /dev/media0
```
